m3/config.go

```diff
@@ -45,6 +45,9 @@ type Configuration struct {
 
 	// IncludeHost is whether or not to include host tag.
 	IncludeHost bool `yaml:"includeHost"`
+
+	// SanitizeMetrics is whether or not metric identifiers (name, tags) are sanitized.
+	SanitizeMetrics bool `yaml:"sanitizeMetrics"`
```

Comment:
Instead of making this a `bool` perhaps it would be preferable to expose `sanitizeFn` as a public type and make this a `SanitizeFn`. The default sanitizer could be a noop function, but we could also provide a few different sanitizer functions that correspond to different reporters. For example, a statsd reporter would only replace the characters that are used as delimiters in the statsd protocol and the current sanitizer could be an alphanumeric sanitizer.